runtime/src/stakes.rs

```diff
@@ -227,22 +227,51 @@ impl Stakes<StakeAccount> {
     /// cached.
     pub(crate) fn new<F>(stakes: &Stakes<Delegation>, get_account: F) -> Result<Self, Error>
     where
-        F: Fn(&Pubkey) -> Option<AccountSharedData>,
+        F: Fn(&Pubkey) -> Option<AccountSharedData> + Sync,
     {
-        let stake_delegations = stakes.stake_delegations.iter().map(|(pubkey, delegation)| {
-            let Some(stake_account) = get_account(pubkey) else {
-                return Err(Error::StakeAccountNotFound(*pubkey));
-            };
-            let stake_account = StakeAccount::try_from(stake_account)?;
-            // Sanity check that the delegation is consistent with what is
-            // stored in the account.
-            if stake_account.delegation() == *delegation {
-                Ok((*pubkey, stake_account))
-            } else {
-                Err(Error::InvalidDelegation(*pubkey))
-            }
-        });
+        // im::HashMap doesn't support rayon so we manually build a temporary vector. Note this is
+        // what std HashMap::par_iter() does internally too.
+        let stake_delegations_vec = stakes.stake_delegations.iter().collect::<Vec<_>>();
```

Comment:
I wouldn't define this extra `..._vec` local variable, and just chain remaining steps of pipeline.
or at least just name `stake_delegations` as is the convention in rust.

Reply:
chained